src/python/pants/backend/shell/util_rules/shell_command_test.py

```diff
@@ -197,6 +198,47 @@ def test_chained_shell_commands(rule_runner: RuleRunner) -> None:
                 )
                 """
             ),
+            "src/b/BUILD": dedent(
+                """\
+                experimental_shell_command(
+                  name="msg",
+                  tools=["cp", "echo"],
+                  output_files=["msg"],
+                  command="echo 'shell_command:b' >> msg",
+                  execution_dependencies=["src/a:msg"],
+                )
+                """
+            ),
+        }
+    )
+
+    assert_shell_command_result(
+        rule_runner,
+        Address("src/a", target_name="msg"),
+        expected_contents={"msg": "shell_command:a\n"},
+    )
+
+    assert_shell_command_result(
+        rule_runner,
+        Address("src/b", target_name="msg"),
+        expected_contents={"msg": "shell_command:a\nshell_command:b\n"},
+    )
+
+
+def test_chained_shell_commands_with_workdir(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "src/a/BUILD": dedent(
+                """\
+                experimental_shell_command(
+                  name="msg",
+                  tools=["echo"],
+                  output_files=["msg"],
```

Comment:
hrrrrrm. No, this appears to be an issue with how `Process` produces its outputs: the `Digest` is rooted at the working directory, so _any_ use of `..` will "escape the working directory".

The "good" news is that with this change you would be in a position to _not_ set `workdir`, and instead perform a `cd` in your `command=` value, and be able to specify `output_files` relative to the build root. Something like the following:

```
experimental_shell_command(
  command=f"cd {build_file_dir()} && ./script.sh",
  output_files=[f"{build_file_dir()}/../../file_that_is_elsewhere.txt"],
)
```

Effectively, that's how we'd need to implement `workdir` to solve the problem under the hood, and that would make other things a bit more complicated.

Reply:
discussing this over in Slack (https://pantsbuild.slack.com/archives/C0D7TNJHL/p1673024640921329?thread_ts=1673024121.608969&cid=C0D7TNJHL) because I find the underlying limitation in `Process` somewhat annoying, and there's probably a reasonable fix here.

Reply:
There is, which is to make _every_ `esc` process do literally what I described above. See #17938.